.github/workflows/windows-npm.yml

```diff
@@ -0,0 +1,102 @@
+name: 'Tests on Windows: `nvm install`'
+
+on: [pull_request, push]
+
+env:
+  NVM_INSTALL_GITHUB_REPO: ${{ github.repository }}
+  NVM_INSTALL_VERSION: ${{ github.sha }}
+
+jobs:
+  node_fail_install_git:
+    # Installation with git does not work as the repository contains files not compatible on Windows
+    # !!! IMPORTANT !!!
+    # THIS TEST WILL PASS WITH THE NEXT RELEASE AS THE FILES HAD BEEN RENAMED
```

Comment:
Important thing to notice: my previous PR allowed to test the installation from another repository but only with METHOD=script. When using default git method, it clones the repo to the latest release version. So this tests still fail has there are still some filename not supported on Windows as my previous PR about renaming those files is not part yet of the latest release.
So that brings me questions:
What to do with this test ? Should I remove it ? If not it will have to be updated with next release.
Should I also allow to clone from a changeset passed through NVM_INSTALL_VERSION ?

Reply:
That seems reasonable, yes.

Reply:
My other PR allows to install with git from a given changeset but here it would not work as the changeset we want here is only part of the PR, not available from outside.
So it is a bit tricky to test it properly...

Reply:
The commits on the PR are available by SHA, and you can use `refs/pull/2391` to point to the PR as well, so it should be possible.

Reply:
Yes but it should be taken into account in install_nvm_from_git function to be able to take a branch reference instead of a commit or an SHA. That would imply some significant change, only to support testing it for a PR.
In my point of view, that sounds good to be able to really test everything related to the current PR.

Reply:
I have not been able to checkout the SHA from the PR:

```shell
git clone https://github.com/nvm-sh/nvm.git -b master --depth=1 nvm
cd nvm
git fetch origin refs/pull/2391/merge --depth=1
git checkout -B refs/pull/2391/merge
git log --all --pretty=oneline
# It return this
# 436cda247e1f5f96bf20df8d49f968dd4d3a9523 (grafted, HEAD -> refs/pull/2391/merge, origin/master, origin/HEAD, master) [Fix] remove `nvm_supports_source_options`
git checkout 208342303df9e5df406ffa996f8ca94bbba47ba0
# It returns
# fatal: reference is not a tree: 208342303df9e5df406ffa996f8ca94bbba47ba0
```

So the workaround is to use the install script with the current SHA to check it clones everything properly.
But then I source the nvm.sh from the current PR thanks to the checkout action.
I dont see any other solution.

Reply:
I found what works in fact:
```shell
git fetch origin +208342303df9e5df406ffa996f8ca94bbba47ba0 --depth=1
```